addons/payment_paypal/models/paypal_s2s.py

```diff
@@ -0,0 +1,231 @@
+_logger = logging.getLogger(__name__)
+
+
+class AcquirerPaypalS2s(models.Model):
+    _inherit = 'payment.acquirer'
+
+    # --------------------------------------------------
+    # SERVER2SERVER RELATED METHODS
+    # --------------------------------------------------
+
+    @api.model
+    def paypal_s2s_form_process(self, data):
+        values = {
+            'cc_holder_name': data.get('cc_holder_name'),
+            'cc_number': data.get('cc_number'),
+            'cc_expiry': data.get('cc_expiry'),
+            'cc_cvv': data.get('cc_cvv'),
+            'cc_brand': data.get('cc_brand'),
+            'acquirer_id': int(data.get('acquirer_id')),
+            'partner_id': int(data.get('partner_id'))
+        }
+        payment_method = self.env['payment.method'].sudo().create(values)
+        return payment_method.id
+
+    @api.multi
+    def _paypal_s2s_get_access_token(self):
+        path = '/v1/oauth2/token'
+        token_url = self._get_paypal_urls(self.environment)['paypal_rest_url'] + path
+        credentials = "%s:%s" % (self.paypal_client_id, self.paypal_client_secret)
+        encode_credential = base64.b64encode(
+            credentials.encode('utf-8')).decode('utf-8').replace('\n', '')
+        token_headers = {
+            'Authorization': ('Basic %s' % encode_credential),
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Accept': 'application/json'
+        }
+        token_params = {
+            'grant_type': 'client_credentials',
+        }
+        token_request = urllib2.Request(token_url, urllib.urlencode(token_params), token_headers)
+        token_response = urllib2.urlopen(token_request)
+        token_result = json.loads(token_response.read())
+        token_response.close()
+        return token_result.get('access_token')
+
+
+class TxPaypalS2s(models.Model):
+    _inherit = 'payment.transaction'
+
+    # --------------------------------------------------
+    # SERVER2SERVER RELATED METHODS
+    # --------------------------------------------------
+
+    def _paypal_try_url(self, request, tries=3):
+        done, res = False, None
+        while (not done and tries):
+            try:
+                res = urllib2.urlopen(request)
+                done = True
+            except urllib2.HTTPError as e:
+                res = e.read()
+                e.close()
+                if tries and res and json.loads(res)['name'] == 'INTERNAL_SERVICE_ERROR':
+                    _logger.warning('Failed contacting Paypal, retrying (%s remaining)' % tries)
+            tries = tries - 1
+        if not res:
+            pass
+        result = json.loads(res.read())
+        res.close()
+        return result
+
+    def _paypal_s2s_send(self):
+        path = '/v1/payments/payment'
+        payment_url = self.env['payment.acquirer']._get_paypal_urls(self.acquirer_id.environment)['paypal_rest_url'] + path
+
+        amount_total = self.amount
+        amount = "%0.2f" % float(amount_total)
```

Comment:
The ´amount´ field is already a float with 2 decimals, this conversion is not needed.